src/cache.c

```diff
@@ -9,12 +9,11 @@
  */
 struct cache_entry *alloc_entry(char *path, char *content_type, void *content, int content_length)
 {
-    struct cache_entry *entry = malloc(sizeof(cache_entry));
-    // entry->content = malloc(sizeof(char) * strlen(content));
-    // entry->content = content;
-    entry->path = malloc(sizeof(char) * strlen(path));
+    printf("here1\n");
+    struct cache_entry *entry = malloc(sizeof(struct cache_entry));
+    entry->content = malloc(content_length);
```

Comment:
strings end in a null char, so you almost always have to have +1
`entry->content = malloc(content_length + 1);`

Reply:
The content_length already takes that + 1 into account in the tests. So I assumed it wasn't necessary.